events/filters.py

```diff
@@ -310,6 +310,12 @@ class EventFilter(django_filters.rest_framework.FilterSet):
             "Search for events that are at most as long as the given duration."
         ),
     )
+    start_weekday = django_filters.MultipleChoiceFilter(
```

Comment:
MultipleChoiceFilter defaults to `distinct=True`. Using DISTINCT in LE event listing is a big no-no as the event table in production is long and wide and distinct will result in a big on-disk sort. Parallel on-disk sorts can end up being very slow to complete, if my memory serves me right some queries lasted multiple hours in 2023 before we got rid of all the distincts that were added 2020-2022. 

Now you can't of course just put distinct=False either, as that will result in duplicate rows for multi-day events. The use of `Exists` is probably required. 

The danger of distinct was noted already in 2019: https://github.com/City-of-Helsinki/linkedevents/blame/84f719601606ff84afbe60d7aebcd0d515702cfb/events/api.py#L2458

Reply:
> MultipleChoiceFilter defaults to distinct=True
ahh yas yas my memory had a leak on that side. I'll fix this.